src/core/createEvent.js

```diff
@@ -24,6 +22,11 @@ export default () => {
       content.data = data;
     },
     mergeXdm: createMerger(content, "xdm"),
+    applyCallback(callback) {
+      const xdm = clone(Object(content.xdm));
+      callback(xdm);
+      content.xdm = xdm;
+    },
```

Comment:
This will need some work now that user-provided XDM is merged on serialization: https://github.com/adobe/alloy/pull/258/files#diff-69cd2fefdc76adcc848d33b1263138dbR48-R57

Let me know if you want to talk through how to best approach this.